src/pages/MyMove/PPMBooking/Advance/Advance.test.jsx

```diff
@@ -0,0 +1,278 @@
+      }
+    },
+  );
+
+  it('can toggle optional fields', async () => {
+    render(<Advance />, { wrapper: MockProviders });
+
+    const advanceRequestedYesInput = screen.getByRole('radio', { name: /yes/i });
+    userEvent.click(advanceRequestedYesInput);
+
+    const advanceInput = await screen.findByLabelText('Amount requested');
+    expect(advanceInput).toBeInstanceOf(HTMLInputElement);
+
+    const advanceRequestedNoInput = screen.getByRole('radio', { name: /no/i });
+    userEvent.click(advanceRequestedNoInput);
+
+    await waitFor(() => {
+      expect(screen.queryByLabelText('Amount requested')).not.toBeInTheDocument();
+    });
+  });
+
+  it('routes back to the previous page when the back button is clicked', () => {
+    render(<Advance />, { wrapper: MockProviders });
+
+    const backButton = screen.getByRole('button', { name: /back/i });
+
+    userEvent.click(backButton);
+
+    expect(mockPush).toHaveBeenCalledWith(estimatedIncentivePath);
+  });
+
+  it('calls the patch shipment endpoint when save & continue is clicked', async () => {
+    patchMTOShipment.mockResolvedValueOnce({ id: mockMTOShipment.id });
+
+    render(<Advance />, { wrapper: MockProviders });
+
+    const advance = 4000;
+    const advanceRequestedYesInput = screen.getByRole('radio', { name: /yes/i });
+    await userEvent.click(advanceRequestedYesInput);
+
+    const agreeToTerms = screen.getByLabelText(/I acknowledge/i);
+    userEvent.click(agreeToTerms);
+
+    const advanceInput = screen.getByLabelText('Amount requested');
+    userEvent.type(advanceInput, String(advance));
+
+    const saveButton = screen.getByRole('button', { name: /save & continue/i });
+    expect(saveButton).not.toBeDisabled();
+    userEvent.click(saveButton);
+
+    const expectedPayload = {
+      ppmShipment: {
+        advance: 400000,
+        id: mockMTOShipment.ppmShipment.id,
+      },
+    };
+
+    await waitFor(() =>
+      expect(patchMTOShipment).toHaveBeenCalledWith(mockMTOShipmentId, expectedPayload, mockMTOShipment.eTag),
+    );
+  });
+
+  it('updates the state if shipment patch is successful', async () => {
+    patchMTOShipment.mockResolvedValue(mockPreExistingShipment);
+
+    render(<Advance />, { wrapper: MockProviders });
+
+    const advance = 4000;
+    const advanceRequestedYesInput = screen.getByRole('radio', { name: /yes/i });
+    userEvent.click(advanceRequestedYesInput);
+
+    const agreeToTerms = screen.getByLabelText(/I acknowledge/i);
+    userEvent.click(agreeToTerms);
+
+    const advanceInput = screen.getByLabelText('Amount requested');
+    userEvent.type(advanceInput, String(advance));
+
+    const saveButton = screen.getByRole('button', { name: /save & continue/i });
+    userEvent.click(saveButton);
+
+    await waitFor(() => expect(mockDispatch).toHaveBeenCalledWith(updateMTOShipment(mockPreExistingShipment)));
+  });
+
+  it('routes to the estimated incentive page when the user clicks save & continue', async () => {
```

Comment:
Small tweak to match expected path
```suggestion
  it('routes to the review page when the user clicks save & continue', async () => {
```